File: Examples/test-suite/octave/callback_runme.m
_callback
callback

if (foo(2) != 2)
  error("failed");
endif

if (A_bar(2) != 4)
  error("failed");
endif

if (foobar(3, _callback.foo) != foo(3))
  error("failed");
endif

if (foobar(3, foo) != foo(3))
  error("failed");
endif

if (foobar(3, A_bar) != A_bar(3))
  error("failed");
endif

if (foobar(3, foof) != foof(3))
  error("failed");
endif

if (foobar_i(3, foo_i) != foo_i(3))
  error("failed");
endif


if (foobar_d(3.5, foo_d) != foo_d(3.5))
  error("failed");
endif

a = A();
if (foobarm(3, a, A.foom_cb_ptr) != a.foom(3))
  error("failed");
endif

File: Examples/test-suite/python/callback_runme.py
import _callback
from callback import *

# callbacks are implemented by modifying docstrings, useful for debugging:
# print("A_bar doc: {}".format(A_bar.__doc__))
# print("A.bar doc: {}".format(A.bar.__doc__))

if foo(2) != 2:
    raise RuntimeError

if A.bar(2) != 4:
    raise RuntimeError

if foobar(3, _callback.foo) != foo(3):
    raise RuntimeError

if foobar(3, foo) != foo(3):
    raise RuntimeError

if foobar(3, A.bar) != A.bar(3):
    raise RuntimeError

if foobar(3, foof) != foof(3):
    raise RuntimeError

if foobar_i(3, foo_i) != foo_i(3):
    raise RuntimeError


if foobar_d(3.5, foo_d) != foo_d(3.5):
    raise RuntimeError

a = A()
if foobarm(3, a, A.foom_cb_ptr) != a.foom(3):
    raise RuntimeError
